contracts/V1/DataTypes.sol

```diff
@@ -11,33 +11,23 @@ contract DataTypes {
         JoinChannelAgreement memory f,
         ConfirmedJoinChannelAgreement memory g,
         LeaveChannel memory h,
-        LeaveChannelAgreement memory i,
-        ConfirmedBlock memory j
+        LeaveChannelAgreement memory i
+       
     ) {}
 }
-//TODO? - think should post state - everyone should be able to replicate the state since genesis (fork) and if a block is posted in the future and some are missing - someone will be folded before the posted BLOCK, as for posting too much in the future it can be challenged
 struct BlockCalldata {
     SignedBlock signedBlock;
     uint timestamp;
 }
-//TODO! - need to rename and refactor this
-struct ForkDataAvailability {
-    mapping(uint => mapping(address => BlockCalldata)) map; //map[transactionCnt][participant] = BlockCalldata
-    ForkDataAvailabilityKey[] keys;
-}
-struct ForkDataAvailabilityKey {
-    uint transactionCnt;
-    address participant;
-}
 
 struct SignedBlock {
-    bytes encodedBlock; //TODO! change this to bytes
-    bytes signature; //TODO! change this to bytes
+    bytes encodedBlock;
+    bytes signature;
 }
 
-struct ConfirmedBlock {
-    bytes encodedBlock;
-    bytes[] signatures; //TODO! change this to bytes
+struct BlockConfirmation {
+    SignedBlock signedBlock;
+    bytes[] signatures;
 }
 
 struct Block {
```

Comment:
The Block struct needs to be updated:
stateHash -> stateSnapshotHash
previousStateHash -> previousBlockHash